src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -0,0 +1,217 @@
+            {
+                this.RegisterConverters(converters, primitiveConverters);
+            }
+        }
+
+        private void RegisterConverters(IEnumerable<JavaScriptConverter> javaScriptConverters, IEnumerable<JavaScriptPrimitiveConverter> javaScriptPrimitiveConverters)
+        {
+            if (javaScriptConverters != null)
+            {
+                this.RegisterConverters(javaScriptConverters);
+            }
+            if (javaScriptPrimitiveConverters != null)
+            {
+                this.RegisterConverters(javaScriptPrimitiveConverters);
+            }
+        }
+
+        public void RegisterConverters(IEnumerable<JavaScriptConverter> javaScriptConverters)
+        {
+            this.converters.AddRange(javaScriptConverters);
+        }
+
+        public void RegisterConverters(IEnumerable<JavaScriptPrimitiveConverter> javaScriptPrimitiveConverters)
+        {
+            this.primitiveConverters.AddRange(javaScriptPrimitiveConverters);
+        }
+
+        protected override string MapClrMemberNameToJsonFieldName(string clrPropertyName)
+        {
+            return this.retainCasing ? base.MapClrMemberNameToJsonFieldName(clrPropertyName) :
+                clrPropertyName.ToCamelCase();
+        }
+
+        public override object DeserializeObject(object value, Type type)
+        {
+            if (type.IsEnum || (ReflectionUtils.IsNullableType(type) && Nullable.GetUnderlyingType(type).IsEnum))									
+                return value == null ? null : Enum.Parse(ReflectionUtils.IsNullableType(type) ? Nullable.GetUnderlyingType(type) : type, value.ToString(), true);
+
+            var primitiveConverter = this.FindPrimitiveConverter(type);
+            if (primitiveConverter != null)
+            {
+                return primitiveConverter.Deserialize(value, type);
+            }
+            var valueDictionary = value as IDictionary<string, object>;
+            if (valueDictionary != null)
+            {
+                var javascriptConverter = this.FindJavaScriptConverter(type);
+                if (javascriptConverter != null)
+                {
+                    return javascriptConverter.Deserialize(valueDictionary, type);
+                }
+
+                if (type.IsGenericType)
+                {
+                    var genericType = type.GetGenericTypeDefinition();
+                    var genericTypeConverter = this.FindJavaScriptConverter(genericType);
+                    if (genericTypeConverter != null)
+                    {
+                        var values = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+                        var genericArguments = type.GetGenericArguments();
+                        for (int i = 0; i < genericArguments.Length; i++)
+                        {
+                            values.Add(valueDictionary.Keys.ElementAt(i), this.DeserializeObject(valueDictionary.Values.ElementAt(i), genericArguments[i]));

```

Comment:
There is a lot going on in this line, please split it up so it is easier to read